### PR TITLE
feat(openSSF scorecard): introduce CI + badge

### LINKED ISSUE #211: OpenSSF scorecard + openSSF best practice.
OpenSSF scorecard is a super tools done by the OpenSSF to give a security score of an repo.
- [openSSF scorecard docs](https://securityscorecards.dev)

Best practice is giving a tier for the accomplished good open source practice of the project.
- [bestpractices website](https://www.bestpractices.dev/)

### PR DESCRIPTION
**Pull request by AugustinMauroy**
Adding openSSF scorecard. It's add CI and badge. Also action allow us to have an UI to see what is wrong or good.

Fixes #211

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 [![codecov](https://codecov.io/gh/valkey-io/valkey/graph/badge.svg?token=KYYSJAYC5F)](https://codecov.io/gh/valkey-io/valkey)
+[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/valkey-io
+/valkey/badge)](https://securityscorecards.dev/viewer/?uri=github.com/valkey-io
+/valkey)
```
Comment (madolson) [inline README.md]:
```suggestion
[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/valkey-io/valkey/badge)](https://securityscorecards.dev/viewer/?uri=github.com/valkey-io/valkey)
```
Is this supposed to be this?

**Comment by madolson** [comment]
@bjosv Does this seem right to you?

File: README.md (hunk 1 above)
Comment (AugustinMauroy) [reply README.md]: I didn't get your question ?

Comment (madolson) [reply README.md]: I proposed a suggestion, it looks like there is new line which shouldn't be there.

Comment (AugustinMauroy) [reply README.md]: Good catch

File: .github/workflows/scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '25 6 * * 6'
+  push:
+    branches: [ "main" ]
```
Comment (zuiderkwast) [inline .github/workflows/scorecard.yml]: We don't have branch called "main".

**Review by zuiderkwast — COMMENTED** [review]
@AugustinMauroy Can you add it on your fork's unstable branch so we can see that it works there?

File: .github/workflows/scorecard.yml (hunk 2 above)
Comment (bjosv) [inline .github/workflows/scorecard.yml]: Only the default branch is supported, which is
```suggestion
    branches: [ "unstable" ]
```

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
```
Comment (bjosv) [inline .github/workflows/scorecard.yml:30]: Is this needed? The default permission is read-all already.

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecard on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action?tab=readme-ov-file#authentication-with-fine-grained-pat-optional.
+          # repo_token: ${{ secrets.SCORECARD_TOKEN }}
+
+          # Public repositories:
+          #   - Publish results to OpenSSF REST API for easy access by consumers
+          #   - Allows the repository to include the Scorecard badge.
+          #   - See https://github.com/ossf/scorecard-action#publishing-results.
+          # For private repositories:
```
Comment (bjosv) [inline .github/workflows/scorecard.yml:54]: Maybe we should remove comments about private repos here in the valkey repo?

File: .github/workflows/scorecard.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          #   - `publish_results` will always be set to `false`, regardless
+          #     of the value entered here.
+          publish_results: true
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@97a0fba1372883ab732affbe8f94b823f91727db # v3.pre.node20
```
Comment (bjosv) [inline .github/workflows/scorecard.yml:62]: Ossf's examples uses the latest version of actions/upload-artifact, i.e. v4.3.3.
Any reason for using v3?

File: .github/workflows/scorecard.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
+
+      # Upload the results to GitHub's code scanning dashboard (optional).
+      # Commenting out will disable upload of results to your repo's Code Scanning dashboard
+      - name: "Upload to code-scanning"
+        uses: github/codeql-action/upload-sarif@1b1aada464948af03b950897e5eb522f92603cc2 # v3.24.9
```
Comment (bjosv) [inline .github/workflows/scorecard.yml:71]: There is a v3.25.3 now.

**Review by bjosv — COMMENTED** [review]
Nice, but some comments below.

Is this based on a template that I haven't seen, or is taken from https://github.com/ossf/scorecard/blob/main/.github/workflows/scorecard-analysis.yml ?

File: .github/workflows/scorecard.yml (hunk 5 above)
Comment (AugustinMauroy) [reply .github/workflows/scorecard.yml:62]: template i had found use v3 so we can update it

**Pull request closed without merging**